worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/TalonEmailSplitter.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright 2017-2022 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing.worker.markup.core;
+
+import com.google.common.base.CharMatcher;
+import com.google.common.collect.Lists;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A Java port of <a href="https://github.com/mailgun/talon/commits/master/talon/quotations.py">quotations.py</a> from the Talon library.
+ *
+ * This port was created from the master branch of Talon on 25 October 2022.
+ *
+ * Due to a particular <a href="https://github.com/mailgun/talon/commit/8138ea9a604f037f47f544dfb805f13d26696275">commit</a> changing some
+ * behaviour that caused some of our integration tests to fail (see the
+ * <a href="https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=477072">comments on Octane ticket
+ * 477072</a> for details), we are using the <a href="https://github.com/mailgun/talon/blob/v1.4.6/talon/quotations.py#L142">1.4.6</a>
+ * version of the RE_FROM_COLON_OR_DATE_COLON regex.
+ *
+ * If updating this class, note the following <a href="https://stackoverflow.com/a/10492382/12177456">differences</a> between Java regex
+ * and Python regex:
+ *
+ * <ul>
+ * <li>
+ * Java's matcher.matches() (also Pattern.matches( regex, input )) matches the entire string. It has no direct equivalent in Python. The
+ * same result can be achieved by using re.match( regex, input ) with a regex that ends with $.
+ * </li>
+ * <li>
+ * Java's matcher.find() and Python's re.search( regex, input ) match any part of the string.
+ * </li>
+ * <li>
+ * Java's matcher.lookingAt() and Python's re.match( regex, input ) match the beginning of the string.
+ * </li>
+ * </ul>
+ */
+public final class TalonEmailSplitter
+{
+    private TalonEmailSplitter()
+    {
+    }
+
+    private final static Pattern RE_FWD = Pattern.compile(
+        "^[-]+[ ]*Forwarded message[ ]*[-]+\\s*$",
+        Pattern.CASE_INSENSITIVE | Pattern.MULTILINE);
+
+    private final static Pattern RE_ON_DATE_SMB_WROTE = Pattern.compile(
+        "(-*[>]?[ ]?(On|Le|W dniu|Op|Am|Em|På|Den|Vào)[ ].*(,|użytkownik)(.*\\n){0,2}.*(wrote|sent|a écrit|napisał|schreef|verzond|geschreven|schrieb|escreveu|skrev|đã viết):?-*)");
```

Comment:
Let's split this to avoid going over the guideline:
```suggestion
        "(-*[>]?[ ]?(On|Le|W dniu|Op|Am|Em|På|Den|Vào)[ ].*(,|użytkownik)(.*\\n){0,2}.*"
        + "(wrote|sent|a écrit|napisał|schreef|verzond|geschreven|schrieb|escreveu|skrev|đã viết):?-*)");
```